src/main/java/org/apache/commons/io/FileUtils.java

```diff
@@ -1452,9 +1452,39 @@ public static File getTempDirectory() {
         return new File(getTempDirectoryPath());
     }
 
+    /**
+     * Returns the path to the system temporary directory with the option to include or disinclude the trailing 
+     * file separator. 
+     *
+     * @param includeTrailingSeparator switch indicating whether or not you want the returned path to include a 
+     * trailing file separator. 
+     * @return the path to the system temporary directory with or without the trailing file separator as per 
+     * caller supplied argument.
+     * @since 2.14.1
+     */
+    public static String getTempDirectoryPathConsistent(boolean includeTrailingSeparator) {
+        String tempDir = System.getProperty("java.io.tmpdir");
+        String tempDirLastChar = tempDir.substring(tempDir.length() - 1);
+        String systemFileSeparator = System.getProperty("file.separator");
```

Comment:
Just a thought, but is it possible to use `tempDir.charAt(tempDir.length() - 1)` and `File.separatorChar`?

Reply:
what do you mean? you want  ln1468 changed to use `File.separatorChar` ? what would be the difference?

Reply:
It would use `char` instead of `String` for the comparison, which would save some memory. Perhaps it's premature optimization, but whenever I can use primitives instead of objects I go for it.

Reply:
> what do you mean? you want ln1468 changed to use `File.separatorChar` ? what would be the difference?

It's better to use an API than a magic string.

Reply:
I agree - that feels like a premature optimization :smile: